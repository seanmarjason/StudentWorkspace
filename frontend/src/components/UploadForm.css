.uploadForm {
  z-index: 100;
  position: fixed;
  left: 50%;
  transform: translate(-50%, 0);
  background-color: white;
  box-shadow: 0 0 25px rgb(0 0 0 / 20%);
  padding: 50px;
}

.button-container {
  display: flex;
  justify-content: center;
}

.closeButton {
  cursor: pointer;
  position: absolute;
  top: 0%;
  right: 0%;
  padding: 12px 16px;
}

.closeButton:hover {
  background: rgb(210, 210, 210);
}
