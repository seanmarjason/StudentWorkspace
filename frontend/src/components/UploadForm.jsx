import axios from 'axios';
import { useOutsideClick } from '../hooks/useOutsideClick';
import './UploadForm.css';

function handleFileChange(event, setFile) {
  setFile(event.target.files[0]);
}

function handleUploadSubmit(event, file) {
  event.preventDefault()
  const url = 'http://localhost:3000/documents/upload';
  const formData = new FormData();
  
  formData.append('file', file);
  formData.append('fileName', file.name);

  const config = {
    headers: {
      'content-type': 'multipart/form-data'
    }
  };

  axios.post(url, formData, config)
    .then((response) => {
      console.log(response.data);
    });
}

const UploadForm = ({file, setFile, callback}) => {

  const handleClickOutside = () => {
    callback(false);
  };

  const ref = useOutsideClick(handleClickOutside);

  return (
    <div ref={ref} className="form uploadForm">
      <form onSubmit={(event) => handleUploadSubmit(event, file)}>
        <div className="title">File Upload</div>
        <div className="button-container">
          <input type="file" onChange={(event) => handleFileChange(event, setFile)}/>
          <button type="submit">Upload</button>
          <button onClick={() => callback(false)}>Cancel</button>
          <span className="closeButton" onClick={() => callback(false)}>&#x2715;</span>
        </div>
      </form>
      <hr></hr>
    </div>
    )
  };

export { UploadForm }
