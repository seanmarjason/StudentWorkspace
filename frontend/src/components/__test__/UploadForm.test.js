import {render } from '@testing-library/react';
import { UploadForm } from '../UploadForm';

describe('Upload Form' , () => {
  it('should match snapshot', async () => {
    const view = render(<UploadForm />)

    expect(view.asFragment()).toMatchSnapshot();
  });
})
  