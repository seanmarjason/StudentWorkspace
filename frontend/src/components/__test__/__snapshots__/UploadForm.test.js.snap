// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload Form should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="form uploadForm"
  >
    <span
      class="closeButton"
    >
      âœ•
    </span>
    <form>
      <h4
        class="title"
      >
        File Upload
      </h4>
      <div
        class="button-container"
      >
        <input
          type="file"
        />
        <button
          class="submitButton"
          type="submit"
        >
          Upload
        </button>
      </div>
    </form>
    <p
      class="separator"
    >
      -- OR --
    </p>
    <form>
      <h4
        class="title"
      >
        Add Link to External Service
      </h4>
      <input
        class="urlInput"
        name="name"
        placeholder="My Link Name"
        required=""
        type="string"
        value=""
      />
      <input
        class="urlInput"
        name="url"
        placeholder="https://www.example.com/"
        required=""
        type="url"
      />
      <button
        class="submitButton"
        type="submit"
      >
        Upload
      </button>
    </form>
    <hr />
    <div
      class="uploadingStatus"
    />
  </div>
</DocumentFragment>
`;
